content/realtime/channels/channel-parameters/deltas.textile

```diff
@@ -0,0 +1,207 @@
+
+
+```[jsall]
+  const vcdiffPlugin = require('{{vcdiff-plugin}}')
+  const realtime = new Ably.Realtime({key: '{{API_KEY}}', plugins: {vcdiffDecoder: vcdiffDecoder}});
+  realtime.channels.get('{{RANDOM_CHANNEL_NAME}}', {
+    delta: 'vcdiff'
+  }).subscribe(msg => console.log("Received message: ", msg));
+```
+
+```[java]
+  AblyRealtime ably = new AblyRealtime("{{API_KEY}}"")
+  Channel channel = ably.channels.get("{{RANDOM_CHANNEL_NAME}}", new ChannelOptions{{params = Map.of("delta", "vcdiff")}});
+  channel.subscribe(new MessageListener() {
+    @Override
+    public void onMessage(Message message) {
+      System.out.println("Received `" + message.name + "` message with data: " + message.data);
+    }
+  });
+```
+
+```[obj-c]
+TBD
+```
+
+```[dotnet]
+TBD
+```
+
+h3(#using-deltas-non-ably). Via a subscription that does not use an Ably library
+
+If subscribing to a channel in delta mode using "SSE":/sse or one of the protocol adaptors such as "MQTT":/mqtt, then you will need to decode any received delta messages yourself. There are decoder libraries available to do this for several platforms; see the "download":https://www.ably.io/download section for details. If you need to decode @vcdiff@-formatted delta messages in languages for which there is no decoder available from Ably, then any compliant open-source implementation of the standard will work.
+
+Messages that contain a @vcdiff@ delta have that delta payload in their @data@ attribute; in the case of @vcdiff@ deltas this will be a binary value, even if the @data@ in the original meessage was text. in addition, there is metadata in the message @extras@ attribute that indicate that the payload is a delta, and which message the delta is relative to. The format of the `extras` attribute is as follows:
+
+```[jsall]
+extras: {
+  format: 'vcdiff',
+  from: '{{previous message id}}'
+}
+```
+
+In order to reconstruct the original message, the @vcdiff@ decoder algorithm needs to be applied to the given @data@ @vcdiff@ value, together with the @data@ for the previous message (on the assumption that that message had already been decoded if it itself was also a delta). If the original form of the present message was text, then this is indicated in the `encoding` attribute of the message, so the original text can be reconstructed by @utf-8@ decoding, once the @vcdiff@ decoding is complete. The decoder libraries that are available to download simplify this process, and more detailed information can be found in the @README@ of each of those libraries.
+
+When subscribing without an Ably library, then the channel @delta@ parameter must be specified using a "qualified channel name":/realtime/channels/channel-parameters/overview. In the case of SSE, it is also possible to specify channel parameters as regular query parameters on the connection URL.
```

Comment:
I'm not sure I understand this question, does SSE not support deltas? cc @paddybyers

Reply:
A couple of issues are confused here.

a) If you're using SSE or any other transport that does not have an Ably library (Pubnub, MQTT) then you can subscribe for deltas by specifying the channel param using a qualified channel name.

b) In the case of SSE only, you can include the channel param as a query param on the connection URL (ie not in the channel name). Then that param will apply to all channels. That is, you can do:

    GET /events?channels=mychannel&delta=vcdiff

instead of:

    GET /events?channels=[?delta=vcdiff]mychannel

c) If you're using the Ably protocol, but are using a library that does not yet support 1.2 and programmatic channel params, then you can also specify the channel param by qualifying the channel name.

Reply:
See https://www.ably.io/download/sdk-feature-support-matrix.  Not all SDKs support all features, and the Delta release should update that feature sheet.

Reply:
I think we should rely on that feature matrix sheet to inform users what is and isn't supported then? @tomczoink how have we dealt with things like this in the past?

Reply:
For the matrix, what're we wanting to consider a feature, and with what releases? Is it just `ChannelParams` with 1.2? Or Deltas at some release and available to all libs (as you can use `[?deltas=vcdiff]` to use them in any), and rewind for the same reason? Or all three? I'm happy to update it to whichever, I'm just not sure how we're considering categorizing what part's a feature and at what stage a feature is part of a lib. @mattheworiordan @paddybyers

Reply:
So, I have actually updated the Feature Matrix with Delta's for now.

ChannelParams doesn't sound to me like a "feature" (user will not choose us over a competitor because of this), it is more of improvement of the develop experience. 

FYI, here is the PR for the feature matrix: https://github.com/ably/website/pull/3227

Reply:
Matrix has been updated by @kavalerov and i have updated the channel parameters overview page to discuss the supported ways in which channel parameters can be sent.